skyportal/facility_apis/atlas.py

```diff
@@ -0,0 +1,272 @@
+
+        target = {
+            'ra': request.obj.ra,
+            'dec': request.obj.dec,
+            'mjd_min': mjd_min,
+            'mjd_max': mjd_max,
+            'send_email': False,
+        }
+
+        return target
+
+
+class ATLASAPI(FollowUpAPI):
+
+    """An interface to ATLAS forced photometry."""
+
+    @staticmethod
+    def get(request):
+
+        """Get a forced photometry request result from ATLAS.
+
+        Parameters
+        ----------
+        request: skyportal.models.FollowupRequest
+            The request to add to the queue and the SkyPortal database.
+        """
+
+        from ..models import (
+            DBSession,
+            FollowupRequest,
+            FacilityTransaction,
+            Allocation,
+            Instrument,
+        )
+
+        req = (
+            DBSession()
+            .query(FollowupRequest)
+            .filter(FollowupRequest.id == request.id)
+            .one()
+        )
+
+        instrument = (
+            Instrument.query_records_accessible_by(request.requester)
+            .join(Allocation)
+            .join(FollowupRequest)
+            .filter(FollowupRequest.id == request.id)
+            .first()
+        )
+
+        altdata = request.allocation.altdata
+
+        if not altdata:
+            raise ValueError('Missing allocation information.')
+
+        content = req.transactions[0].response["content"]
+        content = json.loads(content)
+
+        r = requests.get(
+            content["url"],
+            headers={
+                'Authorization': f"Token {altdata['api_token']}",
+                'Accept': 'application/json',
+            },
+        )
+        r.raise_for_status()
+
+        if r.status_code == 200:
+            if r.json()['finishtimestamp']:
+                result_url = r.json()['result_url']
+                request.status = (
+                    f"Task is complete with results available at {result_url}"
+                )
+
+                s = requests.get(
+                    result_url,
+                    headers={
+                        'Authorization': f"Token {altdata['api_token']}",
+                        'Accept': 'application/json',
+                    },
+                )
+                s.raise_for_status()
+                df = pd.read_csv(
+                    StringIO(s.text.replace("###", "")), delim_whitespace=True
+                )
+                df.rename(
+                    columns={
+                        'MJD': 'mjd',
+                        'RA': 'ra',
+                        'Dec': 'dec',
+                        'm': 'mag',
+                        'dm': 'magerr',
+                        'mag5sig': 'limiting_mag',
+                        'F': 'filter',
+                    },
+                    inplace=True,
+                )
+                cyan = df['filter'] == 'c'
+                orange = df['filter'] == 'o'
+                snr = df['uJy'] / df['duJy'] < 5
```

Comment:
how will this behave if there are NaNs? Or if `df['duJy'] == 0`?

Reply:
Tried some handling of this.